File: app/src/main/java/com/bardahl/maroc/util/ExcelExporter.kt
package com.bardahl.maroc.util

import android.content.Context
import com.bardahl.maroc.domain.model.Order
import java.io.File
import java.io.FileWriter

object ExcelExporter {

    fun exportOrdersToCsv(context: Context, orders: List<Order>): File {
        val file = File(context.cacheDir, "Bardahl_Export_Commandes_${System.currentTimeMillis()}.csv")
        FileWriter(file).use { writer ->
            // Write CSV Header
            writer.append("N° Commande,Date,Commercial,Client,Statut,Total HT (DH),TVA (DH),Total TTC (DH)\n")

            // Write Data Rows
            orders.forEach { order ->
                writer.append("\"${order.orderNumber}\",")
                writer.append("\"${order.orderDate}\",")
                writer.append("\"${order.commercialName}\",")
                writer.append("\"${order.clientName}\",")
                writer.append("\"${order.status.label}\",")
                writer.append("${order.totalHt},")
                writer.append("${order.totalTva},")
                writer.append("${order.totalTtc}\n")
            }
        }
        return file
    }
}

File: react-web/src/utils/excelExporter.js
import * as XLSX from 'xlsx'

export function exportOrdersToExcel(orders) {
  const data = orders.map(o => ({
    "N° Bon": o.orderNumber,
    "Date": o.date,
    "Commercial": o.commercialName,
    "Client": o.clientName,
    "Total HT (DH)": o.totalHt,
    "TVA (DH)": o.totalTva,
    "Total TTC (DH)": o.totalTtc,
    "Statut": o.status
  }))

  const worksheet = XLSX.utils.json_to_sheet(data)
  const workbook = XLSX.utils.book_new()
  XLSX.utils.book_append_sheet(workbook, worksheet, "Commandes")
  XLSX.writeFile(workbook, "Bardahl_Export_Commandes.xlsx")
}
